whelktool/src/main/groovy/whelk/datatool/WhelkTool.groovy

```diff
@@ -150,24 +151,38 @@ class WhelkTool {
         if (!silent) {
             log "Select by ${ids.size()} IDs"
         }
-        def uriIdMap = findShortIdsForUris(ids.findAll { it.contains(':') })
-        def shortIds = ids.findResults { it.contains(':') ? uriIdMap[it] : it }
 
-        def idItems = shortIds.collect { "'$it'" }.join(',\n')
+        def idItems = idLoader.collectXlShortIds(ids).collect { "'$it'" }.join(',\n')
         if (idItems.isEmpty()) {
             return
         }
         doSelectBySqlWhere("id IN ($idItems) AND deleted = false", process,
                 batchSize)
     }
 
+    void selectByIdsAndCollection(Collection<String> ids, String collection, Closure process,
```

Comment:
Do we really need this method? What is the use case?

Reply:
Ok, so collection is needed if we want do do selection by voyager id since they are by collection

Reply:
Voyager-ids are only unique to their collection.